subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/timeout/JavaProcessStackTracesMonitor.java

```diff
@@ -56,10 +56,21 @@ public class JavaProcessStackTracesMonitor {
     private static final Pattern WINDOWS_PID_PATTERN = Pattern.compile("([0-9]+)\\s*$");
     private static final Pattern UNIX_PID_PATTERN = Pattern.compile("([0-9]+)");
 
-    private static PrintStream output;
+    private final File outputFile;
+    private final PrintStream output;
+
+    public JavaProcessStackTracesMonitor(File dumpDir) {
+        try {
+            String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss"));
+            this.outputFile = new File(dumpDir, timestamp + ".threaddump");
+            this.output = new PrintStream(outputFile);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
 
     public static void main(String[] args) {
-        printAllStackTracesByJstack(args.length == 0 ? new File("") : new File(args[0]));
+        new JavaProcessStackTracesMonitor(new File(args.length == 0 ? System.getProperty("user.dir") : args[0])).printAllStackTracesByJstack();
```

Comment:
```suggestion
        new JavaProcessStackTracesMonitor(new File(args.length == 0 ? "." : args[0])).printAllStackTracesByJstack();
```

Should we use `new File(".")` as we do in other places as well?